### ISSUE TITLE
Uncaught PDOException

### ISSUE THREAD

**Issue by jon-heller** [pantheon-systems/wp-saml-auth#53]
Hello,

I have a WordPress site hosted on Pantheon, with SimpleSAML configured. SimpleSAML is working properly when I test it directly through the web interface. 

I've installed and configured this plugin, including adjusting the path to point to the recommended SimpleSAML installation path (`/code/private/simplesaml-1.xx.xx`). However when I click to log in using it, I get the following error:

```
Fatal error: Uncaught PDOException: SQLSTATE[HY000] [2002] No such file or directory in /srv/bindings/43da63f89af94719b907b0f2e2974440/code/private/simplesamlphp-1.14.11/lib/SimpleSAML/Store/SQL.php:54 Stack trace: #0 /srv/bindings/43da63f89af94719b907b0f2e2974440/code/private/simplesamlphp-1.14.11/lib/SimpleSAML/Store/SQL.php(54): PDO->__construct('mysql:host=;por...', NULL, NULL) #1 /srv/bindings/43da63f89af94719b907b0f2e2974440/code/private/simplesamlphp-1.14.11/lib/SimpleSAML/Store.php(49): SimpleSAML_Store_SQL->__construct() #2 /srv/bindings/43da63f89af94719b907b0f2e2974440/code/private/simplesamlphp-1.14.11/lib/SimpleSAML/SessionHandler.php(121): SimpleSAML_Store::getInstance() #3 /srv/bindings/43da63f89af94719b907b0f2e2974440/code/private/simplesamlphp-1.14.11/lib/SimpleSAML/SessionHandler.php(39): SimpleSAML_SessionHandler::createSessionHandler() #4 /srv/bindings/43da63f89af94719b907b0f2e2974440/code/private/simplesamlphp-1.14.11/lib/SimpleSAML/Session.php(148): SimpleSAML_SessionHandler::getSessionHandler() #5 /s in /srv/bindings/43da63f89af94719b907b0f2e2974440/code/private/simplesamlphp-1.14.11/lib/SimpleSAML/Store/SQL.php on line 54
```

I also installed and enabled `Native PHP Sessions for WordPress`. Interestingly, when I add this line to my `www/_include.php` file

`require_once dirname( dirname( dirname( dirname( __FILE__ ) ) ) ). '/wp-load.php';`

I receive the above error trying to simply view my site. 

Out of curiosity I tried setting the SimpleSAML store to `phpsession`, which fixed this error and allowed me to login, but attempted to redirect me back to port 18319 which I assume means some unrelated issue with trying to use `phpsession`.

Thank you!

**Comment by ari-gold** [comment]
Hi @jon-heller -- It looks like this could be a configuration issue, but hard to tell without looking at the codebase. Can you please contact Pantheon support?

**Comment by jon-heller** [comment]
@ari-gold Found the issue!

The [Pantheon instructions on setting up SimpleSAML](https://pantheon.io/docs/shibboleth-sso/), which admittedly are for Drupal, reference getting the Pantheon database info using this code:

```
$ps = json_decode($_SERVER['PRESSFLOW_SETTINGS'], TRUE);
```

However, at least in our instance, we had to retrieve the database settings using the `$_ENV` variable i.e. in SimpleSAML's config.php we used this line

```
'store.sql.dsn'  => 'mysql:host='. $_ENV['DB_HOST'] .';port='. $_ENV['DB_PORT'] .';dbname='. $_ENV['DB_NAME']
```

So it did turn out to be a Pantheon specific configuration issue and not related to this plugin.
